Install fails on macOS

Running `npm install` in an empty project yields the following output:

❯❯❯ sw_vers
ProductName:		macOS
ProductVersion:		15.5
BuildVersion:		24F74

❯❯❯ cmake --version
cmake version 4.0.3

❯❯❯ cat package.json
```json
{
  "name": "jsonnet",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@hanazuki/node-jsonnet": "^3.0.0"
  }
}
```

❯❯❯ npm install
```
npm WARN deprecated npmlog@6.0.2: This package is no longer supported.
npm WARN deprecated are-we-there-yet@3.0.1: This package is no longer supported.
npm WARN deprecated gauge@4.0.4: This package is no longer supported.
npm ERR! code 1
npm ERR! path /Users/***/work/node-jsonnet/node_modules/@hanazuki/node-jsonnet
npm ERR! command failed
npm ERR! command sh -c cmake-js build --CDCMAKE_EXPORT_COMPILE_COMMANDS=1
npm ERR! Not searching for unused variables given on the command line.
npm ERR! -- The C compiler identification is AppleClang 17.0.0.17000013
npm ERR! -- The CXX compiler identification is AppleClang 17.0.0.17000013
npm ERR! -- Detecting C compiler ABI info
npm ERR! -- Detecting C compiler ABI info - done
npm ERR! -- Check for working C compiler: /usr/bin/cc - skipped
npm ERR! -- Detecting C compile features
npm ERR! -- Detecting C compile features - done
npm ERR! -- Detecting CXX compiler ABI info
npm ERR! -- Detecting CXX compiler ABI info - done
npm ERR! -- Check for working CXX compiler: /usr/bin/c++ - skipped
npm ERR! -- Detecting CXX compile features
npm ERR! -- Detecting CXX compile features - done
npm ERR! -- Configuring incomplete, errors occurred!
npm ERR! info TOOL Using Unix Makefiles generator.
npm ERR! info CMD CONFIGURE
npm ERR! info RUN [
npm ERR! info RUN   'cmake',
npm ERR! info RUN   '/Users/***/work/node-jsonnet/node_modules/@hanazuki/node-jsonnet',
npm ERR! info RUN   '--no-warn-unused-cli',
npm ERR! info RUN   '-G',
npm ERR! info RUN   'Unix Makefiles',
npm ERR! info RUN   '-DCMAKE_JS_VERSION=7.3.1',
npm ERR! info RUN   '-DCMAKE_BUILD_TYPE=Release',
npm ERR! info RUN   '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/Users/***/work/node-jsonnet/node_modules/@hanazuki/node-jsonnet/build/Release',
npm ERR! info RUN   '-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded$<$<CONFIG:Debug>:Debug>',
npm ERR! info RUN   '-DCMAKE_JS_INC=/Users/***/work/node-jsonnet/node_modules/node-api-headers/include;/Users/***/work/node-jsonnet/node_modules/node-addon-api',
npm ERR! info RUN   '-DCMAKE_JS_SRC=',
npm ERR! info RUN   '-DNODE_RUNTIME=node',
npm ERR! info RUN   '-DNODE_RUNTIMEVERSION=20.12.2',
npm ERR! info RUN   '-DNODE_ARCH=arm64',
npm ERR! info RUN   '-DCMAKE_OSX_ARCHITECTURES=arm64',
npm ERR! info RUN   '-DCMAKE_EXPORT_COMPILE_COMMANDS=1',
npm ERR! info RUN   '-DCMAKE_JS_LIB=',
npm ERR! info RUN   '-DCMAKE_CXX_FLAGS=-D_DARWIN_USE_64_BIT_INODE=1 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DBUILDING_NODE_EXTENSION',
npm ERR! info RUN   '-DCMAKE_SHARED_LINKER_FLAGS=-undefined dynamic_lookup'
npm ERR! info RUN ]
npm ERR! CMake Error at third_party/jsonnet/CMakeLists.txt:1 (cmake_minimum_required):
npm ERR!   Compatibility with CMake < 3.5 has been removed from CMake.
npm ERR!
npm ERR!   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
npm ERR!   to tell CMake that the project requires at least <min> but has been updated
npm ERR!   to work with policies introduced by <max> or earlier.
npm ERR!
npm ERR!   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
npm ERR!
npm ERR!
npm ERR! ERR! OMG Process terminated: 1

npm ERR! A complete log of this run can be found in: /Users/***/.npm/_logs/2025-07-14T07_08_23_591Z-debug-0.log
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -38,6 +38,10 @@ jobs:
       with:
         path: /home/runner/.cmake-js
         key: ${{ matrix.node }}-cmake-js
+    - name: Set git identity
+      run: |
+        git config --global user.email automate@hanazuki.dev
+        git config --global user.name automate
     - name: Setup compilers
       if: startsWith(matrix.os, 'macos-')
       run: |

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 # ChangeLog
 
+## UNRELEASED
+- The package is now compatible with CMake 4.
+
 ## v3.0.0 (2025-05-08)
 - [breaking] Node.js v14, v16 and v18 are no longer supported.
 - Update libjsonnet to [v0.21.0](https://github.com/google/jsonnet/releases/tag/v0.21.0).

--- a/package.json
+++ b/package.json
@@ -53,6 +53,7 @@
   ],
   "scripts": {
     "install": "cmake-js build --CDCMAKE_EXPORT_COMPILE_COMMANDS=1",
+    "prepare": "if test -f ./scripts/apply-patches.sh; then ./scripts/apply-patches.sh; fi",
     "test": "jasmine",
     "lint": "dtslint types",
     "format": "clang-format -i src/*",

--- a/patches/jsonnet/0001-Add-cmake4-to-cmake_minimum_required.patch
+++ b/patches/jsonnet/0001-Add-cmake4-to-cmake_minimum_required.patch
@@ -1,0 +1,42 @@
+From 6c87c1b0e1e18d25898be071c1b231e264f05a8c Mon Sep 17 00:00:00 2001
+From: Pat Riehecky <riehecky@fnal.gov>
+Date: Wed, 16 Jul 2025 08:58:46 -0500
+Subject: [PATCH] Add cmake4 to cmake_minimum_required.
+
+Leaving cmake 2.8 may have some long term cost, but for now
+it all works as expected.
+
+External projects will need to manage their own cmake_minimum_required.
+
+Signed-off-by: Pat Riehecky <riehecky@fnal.gov>
+---
+ CMakeLists.txt    | 2 +-
+ CMakeLists.txt.in | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index dd17367b..3413ea14 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,4 +1,4 @@
+-cmake_minimum_required(VERSION 2.8.7)
++cmake_minimum_required(VERSION 2.8.7...4.0)
+ project(jsonnet C CXX)
+ 
+ include(ExternalProject)
+diff --git a/CMakeLists.txt.in b/CMakeLists.txt.in
+index 41fac662..7f7d802a 100644
+--- a/CMakeLists.txt.in
++++ b/CMakeLists.txt.in
+@@ -1,7 +1,7 @@
+ # CMake script run a generation-time. This must be separate from the main
+ # CMakeLists.txt file to allow downloading and building googletest at generation
+ # time.
+-cmake_minimum_required(VERSION 2.8.2)
++cmake_minimum_required(VERSION 2.8.2...4.0)
+ 
+ project(googletest-download NONE)
+ 
+-- 
+2.49.0
+

--- a/scripts/apply-patches.sh
+++ b/scripts/apply-patches.sh
@@ -1,0 +1,7 @@
+#!/bin/sh
+set -eux
+
+git submodule update third_party/jsonnet
+for patch in patches/jsonnet/*; do
+  git -C third_party/jsonnet am <"$patch"
+done

